check_run/check_run/doctype/check_run/check_run.py

```diff
@@ -46,6 +46,15 @@ def validate(self):
 				self.set_default_dates()
 		else:
 			self.validate_last_check_number()
+		# Check all selected invoices have correct docstatus (saved/submitted)
+		selected = [txn for txn in json.loads(self.get('transactions')) if txn['pay']]
+		wrong_status = []
+		for t in selected:
+			doc = frappe.get_doc(t['doctype'], t['name'])
+			if doc.docstatus != 1:
+				wrong_status.append(doc.name)
+		if len(wrong_status) > 0:
+			frappe.throw(frappe._('The follow document(s) have been cancelled, please remove them from Check Run to continue: {0}'.format(', '.join(wrong_status))))
```

Comment:
Refactor `format` to an F-string  please. Often a  `<br>`  is a better separator that a comma in these cases, see if you like that better. Easy to read and act on is what we're after.

Reply:
`<br>` tags look good, I'll use those